pkg/solidity-utils/test/foundry/StableMath.t.sol

```diff
@@ -368,4 +371,96 @@ contract StableMathTest is Test {
+        uint256[8] memory balancesRaw
+    ) public view {
+        _testComputeInvariantLessThenInvariantWithDelta(
+            amp,
+            tokenCount,
+            deltaCount,
+            indexes,
+            deltas,
+            balancesRaw,
+            type(uint128).max
+        );
+    }
+
+    function testComputeInvariantLessThenInvariantWithSmallDelta__Fuzz(
+        uint256 amp,
+        uint256 tokenCount,
+        uint256 deltaCount,
+        uint256[8] memory indexes,
+        uint256[8] memory deltas,
+        uint256[8] memory balancesRaw
+    ) public view {
+        _testComputeInvariantLessThenInvariantWithDelta(
+            amp,
+            tokenCount,
+            deltaCount,
+            indexes,
+            deltas,
+            balancesRaw,
+            1000
+        );
+    }
+
+    function _testComputeInvariantLessThenInvariantWithDelta(
+        uint256 amp,
+        uint256 tokenCount,
+        uint256 deltaCount,
+        uint256[8] memory indexes,
+        uint256[8] memory deltas,
+        uint256[8] memory balancesRaw,
+        uint256 maxDelta
+    ) internal view {
+        amp = boundAmp(amp);
+        tokenCount = bound(tokenCount, MIN_TOKENS, MAX_TOKENS);
+        deltaCount = bound(deltaCount, 1, tokenCount);
+
+        uint256[] memory balances = new uint256[](tokenCount);
+        for (uint256 i = 0; i < tokenCount; i++) {
+            balances[i] = bound(balancesRaw[i], 1, type(uint128).max);
+        }
+
+        uint256[] memory newBalances = new uint256[](tokenCount);
+        ScalingHelpers.copyToArray(balances, newBalances);
+
+        for (uint256 i = 0; i < deltaCount; i++) {
+            uint256 tokenIndex = bound(indexes[i], 0, tokenCount - 1);
+            uint256 delta;
+            if (maxDelta > type(uint128).max - newBalances[tokenIndex]) {
+                delta = bound(deltas[i], 0, type(uint128).max - newBalances[tokenIndex]);
+            } else {
+                delta = bound(deltas[i], 0, maxDelta);
+            }
+            newBalances[tokenIndex] += delta;
+        }
+
+        try stableMathMock.computeInvariant(amp, balances, Rounding.ROUND_DOWN) returns (uint256 currentInvariant) {
+            try stableMathMock.computeInvariant(amp, newBalances, Rounding.ROUND_DOWN) returns (
+                uint256 invariantWithDelta
+            ) {
+                if (invariantWithDelta < currentInvariant) {
+                    assertApproxEqAbs(
+                        currentInvariant,
+                        invariantWithDelta,
+                        5,
+                        "Current invariant should be approximately equal to invariant with delta (within 5 wei)"
+                    );
+                } else {
+                    assertLe(
+                        currentInvariant,
+                        invariantWithDelta,
+                        "Current invariant should be less than or equal to invariant with delta"
+                    );
+                }
```

Comment:
```suggestion
                assertLe(
                    currentInvariant - 5,
                    invariantWithDelta,
                    "Current invariant should be less than or equal to invariant with delta (5 wei tolerance)"
                );
```

This also does the trick, it's shorter, but maybe the current version is clearer.

Reply:
I'd say the shorter version is clear enough.

Also, do we need 5 wei? 1 wei works for me. Do lower values fail on CI? (Or it is because we're using 8 tokens, and StableMath is only verified up to 5? Though it seems to work with 1 even with 8 tokens for me.)